crates/database/src/events.rs

```diff
@@ -64,7 +65,8 @@ pub async fn delete(
     ex.execute(sqlx::query(QUERY_TRADE).bind(delete_from_block_number))
         .await?;
 
-    const QUERY_SETTLEMENTS: &str = "DELETE FROM settlements WHERE block_number >= $1;";
+    const QUERY_SETTLEMENTS: &str = "WITH S AS (DELETE FROM settlements WHERE block_number >= $1) \
+                                     DELETE FROM settlement_observations WHERE block_number >= $1;";
```

Comment:
Here you could do 2 queries without needing to refactor anything as you already have a `PgTransaction` available.

Reply:
With this I was aiming to get the atomic execution because my old code assumed that `settlements` and `settlement_observations` tables are in sync always. But I think it is ok to decouple them.